packages/app/src/client/integrations/types.ts

```diff
@@ -14,14 +14,34 @@
  * limitations under the License.
  */
 
-export type IntegrationDef = {
+import { Integration } from "state/integration/types";
+import { Tenant } from "state/tenant/types";
+
+export type FormProps<Data> = {
+  handleCreate: (
+    data: {
+      name: string;
+      data: Data;
+    },
+    options?: { createGrafanaFolder: boolean }
+  ) => void;
+};
+
+export type ShowProps = {}
+
+export type StatusProps = {
+  integration: Integration;
+  tenant: Tenant;
+}
```

Comment:
As all the integration `<Form />`, `<Status />` and `<Show />` were made to have the same interface, I explicitly added that contract as a type.